### ISSUE TITLE
no_std

### ISSUE THREAD

**Issue by stepancheg** [stepancheg/rust-protobuf#610]
Is there a demand for `no_std` feature?

**Comment by Laura7089** [comment]
I'd find that useful!

**Comment by silvergasp** [comment]
I would also find this useful. Happy to put some of my time towards it as well, but I could use some direction.

**Comment by edisno** [comment]
If effort is put into no_std support, please make sure no_alloc and heapless is a part of that solution :)

**Comment by sureshjoshi** [comment]
Are there any Rust protobufs that are heapless/noalloc? Basically, the `nanopb` equivalent?

I'm investigating just embedding nanopb into my embedded app at this point.

**Comment by silvergasp** [comment]
I believe tokio/prost  supports nostd

**Comment by sureshjoshi** [comment]
Yep, they do support no_std. I was testing it out the other day. 

They do require alloc though. 

My (hopefully temporary) solution is to create nanopb bindings using `bindgen` and so far, so good 🤷🏽 

A first party, Rust-native, heapless would be much nicer though.

**Comment by David-OConnor** [comment]
Yes. THat's the only place where I have a use for protobufers. This feels like a canonically no_std task!

**Comment by reneherrero** [comment]
Anyone try this guy? https://github.com/YuhanLiin/micropb

**Comment by David-OConnor** [comment]
Yea; that's what I'm using.
